packages/validators/src/schemas/v2023_09_28/experience.ts

```diff
@@ -0,0 +1,186 @@
+  z.boolean(),
+  z.number(),
+  z.record(z.any(), z.any()),
+  z.undefined(),
+]);
+
+export const ComponentDefinitionPropertyTypeSchema = z.enum([
+  'Text',
+  'RichText',
+  'Number',
+  'Date',
+  'Boolean',
+  'Location',
+  'Media',
+  'Object',
+]);
+
+const ValuesByBreakpointSchema = z.record(z.lazy(() => PrimitiveValueSchema));
+
+const ComponentPropertyValueSchema = z.union([
+  z.object({
+    type: z.literal('DesignValue'),
+    valuesByBreakpoint: ValuesByBreakpointSchema,
+  }),
+  z.object({
+    type: z.literal('BoundValue'),
+    path: z.string(),
+  }),
+  z.object({
+    type: z.literal('UnboundValue'),
+    key: z.string(),
+  }),
+  z.object({
+    type: z.literal('ComponentValue'),
+    key: z.string(),
+  }),
+]);
+
+export type ComponentPropertyValue = z.infer<typeof ComponentPropertyValueSchema>;
+
+const BreakpointSchema = z.object({
+  id: z.string(),
+  query: z.string(),
+  previewSize: z.string(),
+  displayName: z.string(),
+});
+
+const UnboundValuesSchema = z.record(
+  propertyKeySchema,
+  z.object({
+    value: PrimitiveValueSchema,
+  })
+);
+
+const baseComponentTreeNodeSchema = z.object({
+  definitionId: z.string(),
+  variables: z.record(propertyKeySchema, ComponentPropertyValueSchema),
+});
+
+export type ComponentTreeNode = z.infer<typeof baseComponentTreeNodeSchema> & {
+  children: ComponentTreeNode[];
+};
+
+const ComponentTreeNodeSchema: z.ZodType<ComponentTreeNode> = baseComponentTreeNodeSchema.extend({
+  children: z.lazy(() => ComponentTreeNodeSchema.array()),
+});
+
+const ComponentSettingsSchema = z.object({
+  variableDefinitions: z.record(
+    propertyKeySchema,
+    z.object({
+      displayName: z.string(),
+      type: ComponentDefinitionPropertyTypeSchema,
+      defaultValue: ComponentPropertyValueSchema.optional(),
+      description: z.string().optional(),
+      group: z.string().optional(),
+      validations: z.record(z.string()).optional(),
+    })
+  ),
+});
+
+const UsedComponentsSchema = z.array(
+  z.object({
+    sys: z.object({
+      type: z.literal('Link'),
+      id: z.string(),
+      linkType: z.literal('Entry'),
+    }),
+  })
+);
+
+const breakpointsRefinement = (value: Breakpoint[], ctx: z.RefinementCtx) => {
+  if (!value.length || value[0].id !== 'desktop' || value[0].query !== '*') {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: `The first breakpoint should include the following attributes: { "id": "desktop", "query": "*" }`,
+      path: [...ctx.path, 0],
+    });
+  }
+  const queries = value.map((bp) => bp.query);
+  // sort updates queries array in place so we need to create a copy
+  const originalQueries = [...queries];
+  queries.sort((q1, q2) => {
+    if (q1 === '*') {
+      return -1;
+    }
+    if (q2 === '*') {
+      return 1;
+    }
+
+    const q1PixelValue = parseInt(q1.replace(/px|<|>/, ''));
+    const q2PixelValue = parseInt(q2.replace(/px|<|>/, ''));
+
+    return q1PixelValue > q2PixelValue ? -1 : 1;
+  });
```

Comment:
[nitpick] we could extract the values already upfront and then only sort them.
```suggestion
  // Extract the queries boundary by removing the special characters around it
  const queries = value.map((bp) => bp.query === '*' ? bp.query : parseInt(bp.query.replace(/px|<|>/, ''));
  // sort updates queries array in place so we need to create a copy
  const originalQueries = [...queries];
  queries.sort((q1, q2) => {
    if (q1 === '*') {
      return -1;
    }
    if (q2 === '*') {
      return 1;
    }
    return q1 > q2 ? -1 : 1;
  });
```